managed-ledger/src/test/java/org/apache/bookkeeper/test/BookKeeperClusterTestCase.java

```diff
@@ -288,6 +288,7 @@ protected void stopBKCluster() throws Exception {
 
         for (ServerTester t : servers) {
             t.shutdown();
+            t.stopAutoRecovery();
```

Comment:
Doesn't `shutdown` do this? https://github.com/apache/pulsar/blob/32f3577a735581096d85aa961d7df45b9ae9b6f9/testmocks/src/main/java/org/apache/bookkeeper/test/ServerTester.java#L236

Reply:
Yeah, it doesn't work. I need to analyze further why it doesn't work. Thanks for reminder.

Reply:
@horizonzy I did a quick check some time ago and IIRC, it was bugs in the test framework provided by Bookkeeper. There are resource leaks originating from the Bookkeeper test framework.

Reply:
I found that the problem is that the PulsarLedgerAuditorManager auditor election thread is blocked, so the executor can't execute the shutdown task.

The root cause is why the PulsarLedgerAuditorManager election was blocked. In theory, if the leader shuts down, the follower should be the leader, and the election thread shouldn't be blocked.

Reply:
Ok, I found the problem. In the test, it may kill the follower auditor, the follower auditor can't be aware of the close behavior, because the follower auditor is blocked due to waiting leader shutdown.

### The reproduce code:
```
    public TestAutoRecoveryAlongWithBookieServers() throws Exception {
        super(2);
        setAutoRecoveryEnabled(true);
        Class.forName("org.apache.pulsar.metadata.bookkeeper.PulsarMetadataClientDriver");
        Class.forName("org.apache.pulsar.metadata.bookkeeper.PulsarMetadataBookieDriver");
    }


     @Test
    public void testAutoRecoveryAlongWithBookieServers() throws Exception {
        BookieId firstBookie = getBookie(0);
        BookieId secondBookie = getBookie(1);
//        killBookie(firstBookie);
        Auditor auditor = getAuditor(10000, TimeUnit.MILLISECONDS);
        Field bookieIdentifier = auditor.getClass().getDeclaredField("bookieIdentifier");
        bookieIdentifier.setAccessible(true);
        String auditorId = (String) bookieIdentifier.get(auditor);
        System.out.println("kill:" + auditorId);
        if (firstBookie.toString().equals(auditorId)) {
            killBookie(secondBookie);
        } else {
            killBookie(firstBookie);
        }

        Thread.sleep(10000);

        if (firstBookie.toString().equals(auditorId)) {
            killBookie(firstBookie);
        } else {
            killBookie(secondBookie);
        }
    }
```

Reply:
I need further thought to fix it

Reply:
It's hard to fix the problem in the pulsar, fixing it in the bookkeeper would be better. I create a PR in BK to fix it. https://github.com/apache/bookkeeper/pull/4165